src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -210,6 +211,38 @@ def process_response(self, response):
         return response
 
 
+class EmailAddressReplacer(RecordingProcessor):
+    """Replace email address like xxx@microsoft.com with test@example.com"""
+
+    EMAIL_REPLACEMENT = 'test@example.com'
+
+    def _replace_email_address(self, text):
+        pattern = r'[\w.%#+-]+[%40|@|_]microsoft.com'
+        index = 0
+        replaced_text = ''
+        for match in re.finditer(pattern, text):
+            start = match.start()
+            end = match.end()
+            replaced_text += text[index:start] + self.EMAIL_REPLACEMENT
+            index = end
```

Comment:
It it possible to use [`re.sub`](https://docs.python.org/3/library/re.html#re.sub)?

Reply:
We are in a loop for iterating `text` so I'm afraid use `re.sub` to update `text` itself may cause trouble

Reply:
Stupid me🤣